Source/WebCore/animation/WebAnimation.cpp

```diff
@@ -243,14 +243,38 @@ void WebAnimation::setTimeline(RefPtr<AnimationTimeline>&& timeline)
     // 3.4.1. Setting the timeline of an animation
     // https://drafts.csswg.org/web-animations-1/#setting-the-timeline
 
+    // 1. Let old timeline be the current timeline of animation, if any.
+    RefPtr oldTimeline = m_timeline;
+
     // 2. If new timeline is the same object as old timeline, abort this procedure.
-    if (timeline == m_timeline)
+    if (timeline == oldTimeline)
         return;
 
-    // 4. If the animation start time of animation is resolved, make animation's hold time unresolved.
-    if (m_startTime)
-        m_holdTime = std::nullopt;
+    // 3. Let previous play state be animation’s play state (skipped)
+
+    // 4. Let previous current time be the animation’s current time.
```

Comment:
This is split into 2 algorithms in the spec, can we do that here too so it's easier to refer back to the spec?